packages/indexer-common/src/network.ts

```diff
@@ -607,40 +616,48 @@ async function monitorNetworkPauses(
   contracts: NetworkContracts,
   networkSubgraph: NetworkSubgraph,
 ): Promise<Eventual<boolean>> {
-  return timer(60_000)
-    .reduce(async (currentlyPaused) => {
-      try {
-        const result = await networkSubgraph.query(
-          gql`
-            {
-              graphNetworks {
-                isPaused
+  try {
+    return timer(60_000)
+      .reduce(async (currentlyPaused) => {
+        try {
+          const result = await networkSubgraph.query(
+            gql`
+              {
+                graphNetworks {
+                  isPaused
+                }
               }
-            }
-          `,
-        )
+            `,
+          )
 
-        if (result.error) {
-          throw result.error
-        }
+          if (result.error) {
+            throw result.error
+          }
 
-        if (!result.data || result.data.length === 0) {
-          throw new Error(`No data returned by network subgraph`)
-        }
+          if (!result.data || result.data.length === 0) {
+            throw new Error(`No data returned by network subgraph`)
+          }
 
-        return result.data.graphNetworks[0].isPaused
-      } catch (err) {
-        logger.warn(`Failed to check for network pause, assuming it has not changed`, {
-          err: indexerError(IndexerErrorCode.IE007, err),
-          paused: currentlyPaused,
-        })
-        return currentlyPaused
-      }
-    }, await contracts.controller.paused())
-    .map((paused) => {
-      logger.info(paused ? `Network paused` : `Network active`)
-      return paused
+          return result.data.graphNetworks[0].isPaused
+        } catch (err) {
+          logger.warn(`Failed to check for network pause, assuming it has not changed`, {
+            err: indexerError(IndexerErrorCode.IE007, err),
+            paused: currentlyPaused,
+          })
+          return currentlyPaused
+        }
+      }, await contracts.controller.paused())
+      .map((paused) => {
+        logger.info(paused ? `Network paused` : `Network active`)
+        return paused
+      })
+  } catch (error) {
+    logger.error(`Failed to check for network pause with contracts controller`, {
```

Comment:
Could you explain the double wrapped try..catch here? I would expect the inner one to catch the error and still return valid value (`return currentlyPaused`).  Why is the outer try..catch needed if the inner one doesn't throw an error?

Reply:
This is the catch the failure to call  `await contracts.controller.paused()` on [L649](url) when `currentlyPaused` first gets initialized. 
Thinking  definitely be moved to explicitly wrap the contract call. I agree that the inner one doesn't throw an error and easier to follow

Reply:
I left a suggestion for this in #583, which focuses the try/catch scope on the expression that evaluates the network pause.

It is based on this PR.